sdk/communication/azure-communication-callautomation/azure/communication/callautomation/_call_connection_client.py

```diff
@@ -3,272 +3,531 @@
 from ._generated.models import (
-    TransferCallResponse,
+    AddParticipantRequest,
+    RemoveParticipantRequest,
     TransferToParticipantRequest,
+    PlayRequest,
+    RecognizeRequest,
+    ContinuousDtmfRecognitionRequest,
+    SendDtmfRequest,
     CustomContext,
-    AddParticipantRequest,
-    RemoveParticipantResponse,
-    RemoveParticipantRequest
+    DtmfOptions,
+    PlayOptions,
+    RecognizeOptions,
+)
+from ._shared.utils import (
+    get_authentication_policy,
+    parse_connection_str
 )
-
 if TYPE_CHECKING:
-    from ._generated.operations import CallConnectionOperations, CallMediaOperations
-
+    from ._call_automation_client import CallAutomationClient
+    from ._models  import (
+        FileSource,
+        CallInvite
+    )
+    from azure.core.credentials import (
+        TokenCredential,
+        AzureKeyCredential
+    )
+    from ._shared.models import (
+        CommunicationIdentifier,
+    )
+    from ._generated.models._enums import (
+        DtmfTone,
+        RecognizeInputType
+    )
+    from azure.core.exceptions import HttpResponseError
 
 class CallConnectionClient(object): # pylint: disable=client-accepts-api-version-keyword
-    """A client to interact with media of ongoing call.
-
-    :param str call_connection_id:
-     Call Connection Id of ongoing call.
-    :param ~azure.communication.callautomation._generated.operations.CallConnectionOperations call_connection_client:
-     The REST version of call connection client.
-    :param ~azure.communication.callautomation._generated.operations.CallMediaOperations call_media_operations:
-     The REST version of media client.
+    """A client to interact with ongoing call. This client can be used to do mid-call actions,
+    such as Transfer and Play Media. Call must be estbalished to perform these actions.
+
+    :param endpoint: The endpoint of the Azure Communication resource.
+    :type endpoint: str
+    :param credential: The credentials with which to authenticate.
+    :type credential: ~azure.core.credentials.TokenCredential
+     or ~azure.core.credentials.AzureKeyCredential
+    :param call_connection_id: Call Connection ID of ongoing call.
+    :type call_connection_id: str
+    :keyword api_version: Azure Communication Call Automation API version.
+    :paramtype api_version: str
     """
     def __init__(# pylint: disable=missing-client-constructor-parameter-credential, missing-client-constructor-parameter-kwargs
         self,
+        endpoint: str,
+        credential: Union['TokenCredential', 'AzureKeyCredential'],
         call_connection_id: str,
-        call_connection_client,  # type: CallConnectionOperations
-        call_media_operations,  # type: CallMediaOperations
+        *,
+        api_version: Optional[str] = None,
+        **kwargs
     ) -> None:
+        if '_callautomation_client' in kwargs:
+            self._client = cast(AzureCommunicationCallAutomationService, kwargs.pop('_call_automation_client'))
```

Comment:
This should be kwargs.pop('_callautomation_client') otherwise it'll throw a KeyError

Reply:
this part is not needed - now uses: _from_callautomation_client